.github/workflows/consume-release.yml

```diff
@@ -6,13 +6,44 @@ on:
   workflow_run:
     workflows: [Release]
     types: [completed]
+    branches: [develop]
+  workflow_dispatch:
+    inputs:
+      workflowRunId:
+        description: The workflow_run.id of the release workflow that has published artifacts
+        required: true
 
 jobs:
-  on-success:
+  get-images:
+    runs-on: ubuntu-latest
     environment: InfraK8s
+    steps:
+      - id: get-artifacts
+        name: 'Get artifacts from release workflow'
+        uses: actions/github-script@d50f485531ba88479582bc2da03ff424389af5c1 # v6.1.1
+        env:
+          AWS_PRIVATE_ECR_PATH: ${{ secrets.AWS_PRIVATE_ECR_PATH }}
+          WORKFLOW_RUN_ID: ${{ inputs.workflowRunId }}
+        with:
+          result-encoding: string
+          script: |
+            const allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              ...context.repo,
+              run_id: process.env.WORKFLOW_RUN_ID,
+            });
+            // Create whitespace separated URLs for downloading ECR adapter images based on uploaded workflow artifacts 
+            const pathPrefix = process.env.AWS_PRIVATE_ECR_PATH
+            return allArtifacts.data.artifacts
+              .map(({name}) => `${pathPrefix}/adapters/${name.replace('-shasum', '')}`)
+              .join(' ')
+    outputs:
+      published-adapters: ${{ steps.get-artifacts.outputs.result }}
+  image-dispatch:
     runs-on: ubuntu-latest
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
`if: ${{ inputs.workflowRunId || github.event.workflow_run.conclusion == 'success' }}`

We dont want to dispatch if there was a failure on the workflow run